contracts/router/modules/pool/gmx/GMXV1Module.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {IERC20, SafeERC20} from "@openzeppelin/contracts-4.5.0/token/ERC20/utils/SafeERC20.sol";
+
+import {IndexedToken, IPoolModule} from "../../../interfaces/IPoolModule.sol";
+import {IGMXV1Router} from "../../../interfaces/gmx/IGMXV1Router.sol";
+import {IGMXV1Vault} from "../../../interfaces/gmx/IGMXV1Vault.sol";
+
+/// @notice PoolModule for GMX V1 pools
+/// @dev Implements IPoolModule interface to be used with pools addeded to LinkedPool router
+contract GMXV1Module is IPoolModule {
+    using SafeERC20 for IERC20;
+
+    IGMXV1Router public immutable router;
+    IGMXV1Vault public immutable vault;
+
+    constructor(address _router) {
+        router = IGMXV1Router(_router);
+        vault = IGMXV1Vault(router.vault());
+    }
+
+    function poolSwap(
+        address pool,
+        IndexedToken memory tokenFrom,
+        IndexedToken memory tokenTo,
+        uint256 amountIn
+    ) external returns (uint256 amountOut) {
+        // TODO: onlyDelegateCall?
+        require(pool == address(vault), "pool != vault");
+        address tokenIn = tokenFrom.token;
+        IERC20(tokenIn).safeApprove(address(router), amountIn);
+
+        address tokenOut = tokenTo.token;
+        uint256 balanceTo = IERC20(tokenOut).balanceOf(address(this));
+
+        address[] memory path = new address[](2);
+        path[0] = tokenFrom.token;
+        path[1] = tokenTo.token;
+        router.swap(path, amountIn, 0, address(this));
+
+        amountOut = IERC20(tokenOut).balanceOf(address(this)) - balanceTo;
+    }
+
+    function getPoolQuote(
+        address pool,
+        IndexedToken memory tokenFrom,
+        IndexedToken memory tokenTo,
+        uint256 amountIn,
+        bool probePaused
+    ) external view returns (uint256 amountOut) {
+        require(pool == address(vault), "pool != vault");
+        // TODO: simulate like balancer module; would need to remove view
```

Comment:
NOTE: @ChiTimesChi still needs `getPoolQuote` implementation, but waiting for you to give OK to removing `view` on this function in `IPoolModule.sol` interface.

Relevant to balancer pool module implemenation as well given their "sim the swap then revert" approach to quoting.

Reply:
Implementation for quoting would be pretty simple actually in this approach: simply make `poolSwap` on the pool module public, then call in try catch statement in quoter, with explicit revert after obtaining `amountOut` from the pool swap call.

Reply:
GMX exposes quotes via a separate contract:
https://gmxio.gitbook.io/gmx/contracts#swap

Reply:
@ChiTimesChi added 🖖